Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp

```diff
@@ -10072,6 +10075,39 @@ IGNORE_CLANG_WARNINGS_END
         auto* globalObject = m_graph.globalObjectFor(m_origin.semantic);
         setInt32(m_out.castToInt32(vmCall(Int64, operationStringLocaleCompare, weakPointer(globalObject), lowString(m_node->child1()), lowString(m_node->child2()))));
     }
+
+    void compileStringIndexOf()
+    {
+        LValue base = lowString(m_node->child1());
+        LValue search = lowString(m_node->child2());
+        auto* globalObject = m_graph.globalObjectFor(m_origin.semantic);
+        if (m_node->child3()) {
+            std::optional<UChar> character;
+            String searchString = m_node->child2()->tryGetString(m_graph);
+            if (!!searchString) {
+                if (searchString.length() == 1)
+                    character = searchString.characterAt(0);
+            }
+
+            if (character)
+                setInt32(m_out.castToInt32(vmCall(Int64, operationStringIndexOfWithIndexWithOneChar, weakPointer(globalObject), base, lowInt32(m_node->child3()), m_out.constInt32(character.value()))));
+            else
+                setInt32(m_out.castToInt32(vmCall(Int64, operationStringIndexOfWithIndex, weakPointer(globalObject), base, search, lowInt32(m_node->child3()))));
+            return;
+        }
+
+        std::optional<UChar> character;
+        String searchString = m_node->child2()->tryGetString(m_graph);
+        if (!!searchString) {
+            if (searchString.length() == 1)
+                character = searchString.characterAt(0);
+        }
```

Comment:
Why compute this in 2 places?  Why not just do this unconditionally before the `if (m_node->child3())` check above?

Reply:
Changed.